osutil/cp.go

```diff
@@ -117,6 +117,46 @@ func CopyFile(src, dst string, flags CopyFlag) (err error) {
 	return nil
 }
 
+// CopyFileAtomic copies src to dst using AtomicFile internally to create the
+// destination. The destination path is always overwritten. The destination and
+// the owning directory are synced after copy completes. Pass additional flags
+// for AtomicFile wrapping the destination.
+func CopyFileAtomic(src, dst string, flags AtomicWriteFlags) (err error) {
```

Comment:
It slightly bothers me slightly that "CopyFileAtomic" takes different flags than "CopyFile()". OTOH maybe the perfect should not be the enemy of the good and we can always tweak/fix further as needed.

Reply:
The problem is that we cannot really make any use of the flags. CopyFlagSync is meaningless because we always sync,  CopyFlagOverwrite is racy unless we try to open the dst file with O_CREAT|O_EXCL next to AtomicFile, lastly 	CopyFlagPreserveAll cannot be implemented right now without calling `cp` under the hood. In that sense, the semantics are more like AtomicFile, so instead passing CopyFlags, it's taking AtomicWriteFlags.

Reply:
maybe this should be called AtomicCopyFile.  Regarding needeing cp, we could use it if needed and cp to a random file name and then AtomicRename that to the final name, it's not our current use case I think though.

Reply:
The other possible naming would be AtomicWriteFileCopy,  in that case though src and dst would have to be swapped I think

Reply:
I like AtomicWriteFileCopy() and we can always consolidate into CopyFile() with the Atomic flag later, maybe a XXX: should we merge into CopyFIle with new AtomicCopyFile or similar.